src/bitmap/bytes.rs

```diff
@@ -0,0 +1,164 @@
+use crate::bitmap::{bitmask_for_key, index_for_key};
+use crate::{Bitmap};
+#[cfg(feature = "bytes")]
+use bytes::{BufMut, Bytes, BytesMut};
+use std::convert::TryInto;
+
+/// A plain, heap-allocated, `O(1)` indexed bitmap using `bytes::BytesMut` for storage.
+///
+/// This type provide fast O(1) read and write operations, but trades O(n) space complexity for the
+/// additional performance.
+///
+/// The [BytesBitmap] representation is suitable for persistence without the need for serialisation;
+/// the output of [BytesBitmap::freeze()] can be used to construct a new instance. [Serde]
+/// serialisation is also implemented as a conveinence to enable serialisation to various formats.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg(feature = "bytes")]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct BytesBitmap {
+    max_key: usize,
+    bitmap: BytesMut,
+}
+
+#[cfg(feature = "bytes")]
+impl BytesBitmap {
+    pub fn freeze(self) -> Bytes {
+        self.bitmap.freeze()
+    }
+
+    pub fn max_key(&self) -> usize {
+        self.max_key
+    }
+
+    pub fn from_bytes(bitmap: impl Into<Bytes>) -> Self {
+        let bitmap = bitmap.into();
+        Self {
+            max_key: bitmap.len() * 8,
+            bitmap: BytesMut::from(bitmap),
+        }
+    }
+}
+
+#[cfg(feature = "bytes")]
+impl Bitmap for BytesBitmap {
+    fn new_with_capacity(max_key: usize) -> Self {
+        let size = (index_for_key(max_key) + 1) * size_of::<usize>();
+        let bytes = BytesMut::zeroed(size);
+
+        Self {
+            bitmap: bytes,
+            max_key,
+        }
+    }
+
+    fn set(&mut self, key: usize, value: bool) {
+        let offset = index_for_key(key);
+        let byte_offset = offset * size_of::<usize>();
+
+        let slice = &mut self.bitmap[byte_offset..byte_offset + size_of::<usize>()];
+        let mut num = usize::from_ne_bytes(slice.try_into().unwrap());
+
+        if value {
+            num |= bitmask_for_key(key);
+        } else {
+            num &= !bitmask_for_key(key);
+        }
+
+        slice.copy_from_slice(&num.to_ne_bytes());
+    }
+
+    fn get(&self, key: usize) -> bool {
+        let offset = index_for_key(key);
+        let byte_offset = offset * size_of::<usize>();
+        let slice = &self.bitmap[byte_offset..byte_offset + size_of::<usize>()];
+        let num = usize::from_ne_bytes(slice.try_into().unwrap());
+        num & bitmask_for_key(key) != 0
+    }
+
+    fn byte_size(&self) -> usize {
+        self.bitmap.len()
+    }
+    
+    fn or(&self, other: &Self) -> Self {
+        assert_eq!(self.bitmap.len(), other.bitmap.len());
```

Comment:
It's probably best to assert that the inputs are multiples of the chunk size as the code does not attempt to process any remaining bytes, so this acts as a correctness assert:

```suggestion
        assert_eq!(self.bitmap.len(), other.bitmap.len());
        assert_eq!(self.bitmap.len() % size_of::<usize>(), 0);
```

(this always holds, but the assert hardens against future changes)